### PR TITLE
Fix Inconsistent code rendering in documentation examples

### PR DESCRIPTION
**Pull request by Sahil-Shadwal**
**GitHub Issue:** #349 

**Summary**:
The documentation inconsistent code rendering examples have been fixed.

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

### CODE REVIEW

**Comment by DarhkVoyd** [comment]
Thanks for this. I'll review the changes soon. Meanwhile, please link this PR to the original issue.
Reference: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Review by DarhkVoyd — CHANGES_REQUESTED** [review]
Hey, Thank you for this one. Kindly, review on my comments.
1. I believe your editor's markdown linter has changed certain parts of the markdown. Namely, changing `*` to `-` and changing floating point numbers such as `12.50` to `12.5` and some of the indentation. They should not be a part of the PR, kindly revert those.
2. Your approach is mostly valid, but the goal of previous syntax was to demonstrate that the json key is being built upon the previous step's json key or the schema object itself so that it is clearly visible that those are relative and nested changes. This is no longer explicit. We could use some more `...` to make it explicit. 
3. Lastly, if possible show "json" instead of "data".

**Comment by DarhkVoyd** [comment]
@benjagm Since, the solution involves closing the json, I think we still need to add some `...` so that its abundantly clear that we are building on top of certain sections of schema. Also, I am waiting for an update about `data` to `json`.

**Comment by Sahil-Shadwal** [comment]
Okay will revert those .

**Comment by DarhkVoyd** [comment]
@Sahil-Shadwal Let's wait for the maintainers opinion on my suggestions.

**Comment by benjagm** [comment]
Thanks everyone!!

Why changing the format of the list to use '-' instead '*'? That has nothing to do with the issue and doesn't fix anything.

Regarding the '...' that is true @DarhkVoyd . We need to keep them there.

**Comment by Sahil-Shadwal** [comment]
Sorry about * to - it got auto changed when i saved the file , let me correct that . Is there anything else?

**Comment by DarhkVoyd** [comment]
@Sahil-Shadwal Kindly refer to my previous review. Regarding `...`, please add `...` at appropriate places to ensure that it's abundantly clear that we are building on top of certain sections of schema and clearly express its structure. 

> Hey, Thank you for this one. Kindly, review on my comments.
> 
> 1. I believe your editor's markdown linter has changed certain parts of the markdown. Namely, changing `*` to `-` and changing floating point numbers such as `12.50` to `12.5` and some of the indentation. They should not be a part of the PR, kindly revert those.
> 2. Your approach is mostly valid, but the goal of previous syntax was to demonstrate that the json key is being built upon the previous step's json key or the schema object itself so that it is clearly visible that those are relative and nested changes. This is no longer explicit. We could use some more `...` to make it explicit.
> 3. Lastly, if possible show "json" instead of "data".

File: pages/learn/getting-started-step-by-step.md [hunk 1]
```diff
@@ -99,28 +99,32 @@ Using the product catalog example, `productId` is a numeric value that uniquely
 To add the `properties` object to the schema:
 
 1. Add the `properties` validation keyword to the end of the schema:
-    
+
 ```json
-...
-"title": "Product",
```
Comment (gregsdennis) [inline pages/learn/getting-started-step-by-step.md:105]: If we're going to make these syntax-highlighted, we need to do this differently.

    ```jsonc
    // ...
    "title": "Product",

`jsonc` allows comments, and I'm pretty sure we can get it to have the same highlight colors as `json` (probably in another PR if it's not already that way)

File: pages/learn/getting-started-step-by-step.md [hunk 2]
```diff
@@ -490,21 +531,25 @@ The following schema validates a geographical location:
...
-"properties": {
-...
-  "warehouseLocation": {
+{
+  "properties": {
+  ...
+    "warehouseLocation": {
+    }
   }
 }
```
Comment (gregsdennis) [inline pages/learn/getting-started-step-by-step.md:542]: Similarly, this isn't the full schema on its own.  The current snippet (although not proper JSON) demonstrates well that this is an excerpt from the above schema, but the new snippet is a schema by itself.

This is seen in a lot of these changes.

File: pages/learn/getting-started-step-by-step.md [hunk 3]
```diff
@@ -346,59 +375,69 @@ The earlier examples describe a flat schema with only one level. This section de
...
 ```json
-"dimensions": {
-  "type": "object",
-  "properties": {
-    "length": {
-      "type": "number"
-    },
-    "width": {
-      "type": "number"
+{
```
Comment (gregsdennis) [inline pages/learn/getting-started-step-by-step.md:424]: like here

File: pages/learn/getting-started-step-by-step.md [hunk 4]
```diff
@@ -346,59 +375,69 @@ The earlier examples describe a flat schema with only one level. This section de
...
-    "length": {
-      "type": "number"
-    },
-    "width": {
-      "type": "number"
-    },
-    "height": {
-      "type": "number"
+{
+  "dimensions": {
```
Comment (gregsdennis) [inline pages/learn/getting-started-step-by-step.md:403]: here

File: pages/learn/getting-started-step-by-step.md [hunk 5]
```diff
@@ -346,59 +375,69 @@ The earlier examples describe a flat schema with only one level. This section de
...
 ```
 
 2. Define the `type` validation keyword as `object`:
-    
+
 ```json
-"dimensions": {
-  "type": "object",
+{
+  "dimensions": {
```
Comment (gregsdennis) [inline pages/learn/getting-started-step-by-step.md:393]: and here

**Review by gregsdennis — CHANGES_REQUESTED** [review]
I think we need to re-evaluate the importance of having proper JSON.  I'm not sure it's needed.  More important than the snippets being "correct" is that they are understandable.  I think many of these edits serve to confuse the reader rather than clarifying the ideas that the narrative is trying to convey.

**Comment by Sahil-Shadwal** [comment]
Thanks for the review, will surely look into it . But it is getting confusing, should i revert back  the changes ?

**Comment by DarhkVoyd** [comment]
@gregsdennis Hey, I am bit confused. As per my understanding, we agree on:

> The current snippet (although not proper JSON) demonstrates well that this is an excerpt from the above schema, but the new snippet is a schema by itself.

So are we against closing the json as a solution, or the usage of `...` or the entire approach.

**Comment by gregsdennis** [comment]
I'm happy to have the `...`, but the changes look like they're trying to create full JSON values, which isn't right.

It looks like the code blocks already have the language specifier.  I'm guessing the bare `...` is causing them to be rendered as plain text.

I think if we change the language specifier to `jsonc` and use `// ...` it _might_ render right.

Otherwise we'll have to do something like

    ```jsonc
    {
      // ...
      "properties": {
        // ...
        "newProperty": {}
      }
    }
    ```

Then we need to make sure the `jsonc` formatting is the same as the `json` formatting.

**Comment by DarhkVoyd** [comment]
Yes, that's what I meant. You explained it better.

**Comment by benjagm** [comment]
@Sahil-Shadwal is this ready to review? Thanks a lot!

**Comment by gregsdennis** [comment]
@Sahil-Shadwal was this replaced by another PR? Could you link to it, please?

**Comment by DarhkVoyd** [comment]
This was replaced with this PR:  https://github.com/json-schema-org/website/pull/475

**Pull request closed without merging**